examples/tutorials/cnndm_summarization.py

```diff
@@ -149,21 +149,76 @@ def apply_prefix(task, x):
 #
 # We can define a sequence generator to produce an output sequence based on the input sequence provided. This calls on the
 # model's encoder and decoder, and iteratively expands the decoded sequences until the end-of-sequence token is generated
-# for all sequences in the batch. The `greedy_generator` method shown below uses a greedy search (i.e. expands the sequence
-# based on the most probable next word).
+# for all sequences in the batch. The `generate` method shown below uses a beam search to generate the sequences.
 #
 
 from torch import Tensor
 from torchtext.prototype.models import T5Model
 
 
-def greedy_generator(
-    encoder_tokens: Tensor,
-    eos_idx: int,
-    model: T5Model,
-) -> Tensor:
+def beam_search(
+    beam_size: int,
+    step: int,
```

Comment:
`step` here is equivalent to the current length of sequences in `decoder_tokens`. The first time `beam_search` is called, `step=1` because `decoder_tokens` is initialized to have the padding token as the starter token to each sequence.